lib/ecto/changeset/relation.ex

```diff
@@ -95,6 +95,14 @@ defmodule Ecto.Changeset.Relation do
     end
   end
 
+  def cast(%{cardinality: :one} = relation, owner, params, current, on_cast) when is_list(params) do
+    if Keyword.keyword?(params) do
+      cast(relation, owner, Map.new(params), current, on_cast)
```

Comment:
Required the keyword check because we allow keyword lists to be cast to embedded schemas